src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -686,6 +686,90 @@ namespace OpenLoco::Vehicles
         return AirportObjectFlags::acceptsHeavyPlanes;
     }
 
+    // 0x004AFFF3
+    VehicleBogie* VehicleBogie::flipCar()
+    {
+        Vehicle train(head);
+        VehicleBase* precedingCarBody = train.veh2;
+        while (precedingCarBody->nextVehicleComponent()->asBase<VehicleBogie>() != this)
+        {
+            precedingCarBody = precedingCarBody->nextVehicleComponent();
+        }
+
+        VehicleBase* ptr = this; // lvalue required
+        CarComponent firstComponent(ptr);
+        ptr = this; // CarComponent constructor changes value
+        Car car(ptr);
+
+        sfl::static_vector<CarComponent, VehicleObject::kMaxCarComponents> components;
+        for (CarComponent& component : car)
+        {
+            auto frontSprite = component.front->objectSpriteType;
+            component.front->objectSpriteType = component.back->objectSpriteType;
+            component.back->objectSpriteType = frontSprite;
+
+            // silence -Werror=null-dereference
+            auto body = component.body->asBase<VehicleCommon>();
```

Comment:
There is no need to do this. You have access to body var_38. `component.body->var_38 ^= Flags38::isReversed;`